torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -456,11 +522,21 @@ FusionExecutor::GlobalBuffers FusionExecutor::allocGlobalVals(
       continue;
     }
     if (!alloc->zeroInit()) {
-      global_buffers.empty_buffers.push_back(
-          inferAndAlloc(tv, alloc->shape(), expr_eval, options_, false));
+      global_buffers.empty_buffers.push_back(inferAndAlloc(
+          tv,
+          alloc->shape(),
+          kernel_summary.memoized_global_shapes[i - 1],
```

Comment:
nit: `i-1` -> `i` and move `++i` to the end of the loop body.

Reply:
Done.